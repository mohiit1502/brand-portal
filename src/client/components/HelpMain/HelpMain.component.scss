.c-HelpMain {
    // overflow-y: scroll;
    // height: 79vh;
    transition: left 4s ease-in-out;

    &.relative {
        position: relative;
    }

    .absolute {
        position: absolute;
    }

    &.slideout {
        left: -100%;
        // opacity: 0;
    }

    &.slidein {
        left: 0;
        opacity: 1;
    }

    .c-HelpMain__header {
        padding-left: 1.5rem;
        padding-bottom: 1.125rem;
        border-bottom: 1px solid #CCCCCC;
        margin-bottom: 0;
    }

    .c-HelpMain__content {
        overflow-y: scroll;
        height: 75vh;

        .c-HelpMain__content__qContainer {
            padding-top: 1rem;
        }
    }

    .c-HelpMain__partial {
        padding-bottom: 1rem;
    }
}
