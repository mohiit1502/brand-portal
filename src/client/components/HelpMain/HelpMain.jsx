import React from "react";
import PropTypes from "prop-types";
import FaqSingle from "./../FaqSingle";
import "./HelpMain.component.scss";

const HelpMain = ({activeTab, content}) => {

  const activeTabContent = content && content[activeTab];
  const isSingleQuestion = !(activeTabContent && activeTabContent.items);

  const contentRendered = activeTabContent && activeTabContent.items ? activeTabContent.items.map((item, index) => {
    item.id = `${activeTab}-${index}`;
    return <FaqSingle key={`${activeTab}-${index}`} data={item} />;
  }) : <div className="c-HelpMain__content__qContainer"><FaqSingle data={activeTabContent} expandPreState={true} /></div>;

  return (
    <div id={activeTabContent.id} className="c-HelpMain">
      <div className="c-HelpMain__header h4" style={{borderBottom: isSingleQuestion && "none"}}>{activeTabContent.header}</div>
      <div className="c-HelpMain__content">
        {contentRendered}
      </div>
    </div>
  );
};

HelpMain.propTypes = {
  activeTab: PropTypes.string,
  content: PropTypes.object
};

export default HelpMain;
