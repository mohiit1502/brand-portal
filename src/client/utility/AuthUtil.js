import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
