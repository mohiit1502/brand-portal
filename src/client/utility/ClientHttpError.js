export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
