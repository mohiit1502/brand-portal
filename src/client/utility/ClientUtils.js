import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
