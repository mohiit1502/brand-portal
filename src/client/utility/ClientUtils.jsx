import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
