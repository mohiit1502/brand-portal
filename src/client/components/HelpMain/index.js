import HelpMain from './HelpMain.jsx';

export default HelpMain;