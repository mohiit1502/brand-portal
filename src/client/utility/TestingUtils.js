import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
