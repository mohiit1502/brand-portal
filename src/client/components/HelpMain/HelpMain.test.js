import React, {useRef} from "react";
import renderer from "react-test-renderer";
import HelpMain from "./HelpMain";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import {Provider} from "react-redux";
import MockNextContext from "../../../../test/client/utility/MockNextContext";
import {clearKeys, testStore} from "../../utility/TestingUtils";


let props;
let store;

const setUp = (props) => {
  store = testStore({});
  return shallow(props ? <Provider store={store}><MockNextContext pathname="/help"><HelpMain {...props} /> </MockNextContext></Provider>: <HelpMain />);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});



describe("HelpMain", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });
  it("renders without error", () => {
    props = {
      activeTab:"1",
      content: {
        "1":{
          id:""
        }
      },
      goToContactUs: () => {}
    };
    wrapper = setUp(props);

    const tree = toJson(wrapper);
    clearKeys(tree, []);
    expect(tree).toMatchSnapshot();

  });
});
