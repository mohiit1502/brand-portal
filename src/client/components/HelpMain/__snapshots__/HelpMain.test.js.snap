// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpMain renders without error 1`] = `
<MockNextContext
  pathname="/help"
>
  <HelpMain
    activeTab="1"
    content={
      Object {
        "1": Object {
          "id": "",
        },
      }
    }
    goToContactUs={[Function]}
  />
   
</MockNextContext>
`;
